src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -122,6 +135,13 @@ export function getQueryParams({
   kueryNode,
 }: QueryParams) {
   const types = getTypes(mappings, type);
+
+  const normalizedNamespaces = namespaces
+    ? Array.from(
+        new Set(namespaces.map((namespace) => (namespace === '*' ? 'default' : namespace)))
```

Comment:
Why does OSS interpret `*` as `'default'`? Usually OSS objects will all be in the default space, but a third-party spaces plugin could leverages the namespaces array and create objects in other spaces. I would then expect '*' to remove all namespaces filtering.

Reply:
We went back and forth on this: https://github.com/elastic/kibana/pull/67644#discussion_r441055716

My original implementation threw an error if a wildcard made its way here, as I had an invalid fear of that being treated as an actual wildcard during search. We aren't using the DSL in a way that allows for the wildcard, so it would instead be treated as a literal `*` if we passed it through unmodified.

@jportner and I felt that interpreting `*` as default would be the most consistent with the way OSS behaves today, or the way the default distribution behaves when spaces is disabled. I could see an argument for supporting a third-party spaces plugin, but it didn't seem worthwhile to me. I worry about stripping out the namespaces filtering altogether, given the complexity of the saved objects service (I would hate to accidentally expose objects the user isn't authorized to see). This is probably an irrational fear though, so we can explore alternatives if you'd prefer.

Reply:
Can we add a comment explaining the reasoning from https://github.com/elastic/kibana/pull/67644#discussion_r441055716 so we remember why there's this magic 😂